docs/usage/configuration/config_documentation.md

```diff
@@ -845,7 +845,11 @@ which are older than the room's maximum retention period. Synapse will also
 filter events received over federation so that events that should have been
 purged are ignored and not stored again.
 
-The message retention policies feature is disabled by default.
+The message retention policies feature is disabled by default. Please be advised 
+that enabling this feature carries some risk, as it is possible to corrupt your database 
+due to insufficient cache invalidation. Setting retention to delete older history
```

Comment:
Perhaps we could throw the phrase "known bugs" in there? eg. `due to known bugs around cache invalidation`
Insufficient cache invalidation might be a bit opaque to server admins.

Reply:
Though #13476 suggests that not all the bugs are to do with cache invalidation.